cardinal/ecs/options.go

```diff
@@ -42,3 +43,9 @@ func WithStoreManager(s store.IManager) Option {
 		w.storeManager = s
 	}
 }
+
+func WithEventHub(eventHub *events.EventHub) Option {
+	return func(w *World) {
+		w.eventHub = eventHub
+	}
+}
```

Comment:
i think we'd want this to be in world by default, not optional

Reply:
It has to be passed into ecs world externally because this eventHub is something that has to be used by the Server package as well. 

Thus the cardinal layer is where instantiation of `eventHub` occurs. It is "default" in from the cardinal perspective, but from the ecs perspective it is not. We could make it so that ecs.World won't even run without an eventHub, but that complicates every test as now you have to Spawn an unused eventHub

Reply:
ahh gotcha, this is fine then - thanks! 🙏

Reply:
I can make a mock in some later PR. some eventHub that emits things to stdout that doesn't need to be owned by server so we can test stuff.

Reply:
ahh good idea, if you can make a ticket for that that would be sweet